Makefile

```
@@ -177,70 +198,70 @@ clean:
-# Use this to populate the vendor directory after checking out the repository.
-# To update upstream dependencies, delete the glide.lock file first.
-vendor: glide.lock
-	# Ensure that the glide cache directory exists.
-	mkdir -p $(HOME)/.glide
 
-	# To build without Docker just run "glide install -strip-vendor"
+vendor: go.mod go.sum
+	# To build without Docker just run "go mod cache"
 	if [ "$(LIBCALICOGO_PATH)" != "none" ]; then \
-          EXTRA_DOCKER_BIND="-v $(LIBCALICOGO_PATH):/go/src/github.com/projectcalico/libcalico-go:ro"; \
+          EXTRA_DOCKER_ARGS+="-v $(LIBCALICOGO_PATH):/go/src/github.com/projectcalico/libcalico-go:ro"; \
 	fi; \
-
-	docker run --rm \
-		-v $(CURDIR):/go/src/$(PACKAGE_NAME):rw $$EXTRA_DOCKER_BIND \
-		-v $(HOME)/.glide:/home/user/.glide:rw \
-		-e LOCAL_USER_ID=$(LOCAL_USER_ID) \
-		-w /go/src/$(PACKAGE_NAME) \
-		$(CALICO_BUILD) glide install -strip-vendor
+	$(DOCKER_RUN) $(CALICO_BUILD) go mod vendor
 
 ## Default the repos and versions but allow them to be overridden
 LIBCALICO_BRANCH?=$(shell git rev-parse --abbrev-ref HEAD)
 LIBCALICO_REPO?=github.com/projectcalico/libcalico-go
 LIBCALICO_VERSION?=$(shell git ls-remote git@github.com:projectcalico/libcalico-go $(LIBCALICO_BRANCH) 2>/dev/null | cut -f 1)
+LIBCALICO_OLDVER?=$(shell go list -m -f "{{.Version}}" github.com/projectcalico/libcalico-go)
 FELIX_BRANCH?=$(shell git rev-parse --abbrev-ref HEAD)
 FELIX_REPO?=github.com/projectcalico/felix
 FELIX_VERSION?=$(shell git ls-remote git@github.com:projectcalico/felix $(FELIX_BRANCH) 2>/dev/null | cut -f 1)
+FELIX_OLDVER?=$(shell shell go list -m -f "{{.Version}}" github.com/projectcalico/felix)
 CONFD_BRANCH?=$(shell git rev-parse --abbrev-ref HEAD)
 CONFD_REPO?=github.com/projectcalico/confd
 CONFD_VERSION?=$(shell git ls-remote git@github.com:projectcalico/confd $(CONFD_BRANCH) 2>/dev/null | cut -f 1)
+CONFD_OLDVER?=$(shell go list -m -f "{{.Version}}" github.com/projectcalico/confd)
 
-### Update pins in glide.yaml
 update-felix-confd-libcalico:
+ifneq ($(strip $(LIBCALICO_VERSION)),)
+ifneq ($(strip $(LIBCALICO_OLDVER)),)
+	echo "Updating libcalico version $(LIBCALICO_OLDVER) to $(LIBCALICO_VERSION) from $(LIBCALICO_REPO)"
+	go mod edit -droprequire github.com/projectcalico/libcalico-go && go get $(LIBCALICO_REPO)@$(LIBCALICO_VERSION)
+endif
+endif
+ifneq ($(strip $(FELIX_VERSION)),)
+ifneq ($(strip $(FELIX_OLDVER)),)
+	echo "Updating felix version $(FELIX_OLDVER) to $(FELIX_VERSION) from $(FELIX_REPO)"
+	go mod edit -droprequire github.com/projectcalico/felix && go get $(FELIX_REPO)@$(FELIX_VERSION)
+endif
+endif
+ifneq ($(strip $(CONFD_VERSION)),)
+ifneq ($(strip $(CONFD_OLDVER)),)
+	echo "Updating confd version $(CONFD_OLDVER) to $(CONFD_VERSION) from $(CONFD_REPO)"
+	go mod edit -droprequire github.com/projectcalico/felix && go get $(CONFD_REPO)@$(CONFD_VERSION)
+endif
+endif
+	$(DOCKER_RUN) $(CALICO_BUILD) go mod vendor
+
+remote-deps:
+	mkdir -p filesystem/etc/calico/confd/conf.d
```

```suggestion
	mkdir -p filesystem/etc/calico/confd
```

`cp` acts differently depending on whether the target directory exists. The way you have it, it was putting the config files in `filesystem/etc/calico/confd/conf.d/conf.d/` instead of `filesystem/etc/calico/confd/conf.d/`

---

Fixed